### ISSUE TITLE
False positive on SC2191 when using `+=` assignment operator

### ISSUE THREAD

**Issue by Xophmeister** [koalaman/shellcheck#3110]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2191
- My shellcheck version (`shellcheck --version` or "online"): 0.10.0
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/usr/bin/bash

x=(foo bar [1]+=quux)
echo "${x[@]}"
```

#### Here's what shellcheck currently says:

```
In - line 3:
x=(foo bar [1]+=quux)
               ^---^ SC2191 (warning): The = here is literal. To assign by index, use ( [index]=value ) with no spaces. To keep as literal, quote it.

Did you mean:
x=(foo bar [1]+"=quux")

For more information:
  https://www.shellcheck.net/wiki/SC2191 -- The = here is literal. To assign ...
```

#### Here's what I wanted or expected to see:

This shouldn't trigger SC2191.

Note: Only the `+=` operator works in this context, even with `typeset -i`; `++` and `-=` trigger syntax errors in Bash, whereas `--`, `*=` and `/=` get interpreted as literals.

**Comment by wileyhy** [comment]
Um, irc says that syntax could be "cursed."  o_O  ...??

Wiley
